R/derive_param_tte.R

```diff
@@ -374,7 +378,17 @@ derive_param_tte <- function(dataset = NULL,
     )
   }
 
+ #check for duplicates in dataset_adsl and source_datasets
+  combined_dataset <- bind_rows(dataset_adsl, !!!source_datasets)
+
+  signal_duplicate_records(
+    dataset = combined_dataset,
+    by_vars = expr_c(subject_keys, by_vars),
+    cnd_type = check_type
+  )
+
```

Comment:
This is not the right place for the check. Now both `derive_param_tte()` calls in "Test 16" produce a warning. But they shouldn't because `order` is specified there, which makes the selection unique.

I think you can pass `check_type` to `filter_date_sources()` and then pass it to the `filter_extreme()` call. Most likely you need to catch the duplicates message from `filter_extreme()` and rephrase it to make it helpful for the users.